### DISCUSSION TITLE
Prevent file-exec-and-symbols step

### DISCUSSION THREAD

**Discussion by rfon6ngy** [microsoft/vscode-cpptools#11855] [Q&A]
I'm currently trying to connect VSCode to a running (and executing) GDB server,
So I don't want any `exec` step executed by VSCode.
here is my launch.json

```js
{
	"name": "MI GDB Attach",
	"type": "cppdbg",
	"request": "attach",
	"miDebuggerPath": "/path/to/powerpc-eabi-gdb",
	"MIMode": "gdb",
	//"miDebuggerServerAddress": json.gdb_cmd[4] // DO NOT WORK : arrive to late >> do it myself in setup,
	"program": "/path/to/my.elf", // (required) used by file-exec-and-symbols step
	"stopAtConnect": true,
	"processId": 12,
	"useExtendedRemote": true,
	"launchCompleteCommand": "None",
	"setupCommands": [
		{
			"text": "target extended-remote ${json.gdb_cmd[4]}", // value is injected at runtime
			"description": "connect",
			"ignoreFailures": true
		}
	]
}
```
Despite my attempts with `launchCompleteCommand:None` and `stopAtConnect: true`, VSCode tries to run a file-exec-and-symbols which fail in my environement.

I also tried a "request:lauch" kind of run, which get stuck without any logs.

How to prevent this unwanted command to be run by VSCode GDB Client ?

**Comment by jfuentesa** [comment]
Use "request": "launch" with "customLaunchSetupCommands": []
Then, put your server in the "miDebuggerServerAddress".

https://code.visualstudio.com/docs/cpp/launch-json-reference#_customlaunchsetupcommands
https://code.visualstudio.com/docs/cpp/launch-json-reference#_midebuggerserveraddress

With this configuration, the extension is running in "launch" mode and it will kill the process if you STOP the debugger. Instead, use "-exec detach" in the debug console.

It would be great if they add a "customAttachSetupCommands" or something like that...

**Comment by rfon6ngy** [comment]
Thanks for the heads up 👍 
My working solution was to create a wrapper plugin that get the correct values and call `vscode.debug.startDebugging` with the following config:
```js
{
	name: "GDB attach",
	type: "cppdbg",
	request: "attach",
	miDebuggerPath: bin,
	MIMode: "gdb",
	useExtendedRemote: true,
	setupCommands: [
		{ text: `target extended-remote ${address}` },
		{ text: `source ${script_path}` },
	],
	program: debugLine[1],
	processId: +debugLine[2],
}
```

And it work like a charm !

**Reply by jfuentesa** [reply]
Is it possible to pass variables to the configuration? (${address}, ${script_path}...)

**Reply by rfon6ngy** [reply]
not without using an extension (which is what I've created to fill those values)
I know that some extension can offer user dialog by using specific syntax in the config
But in my case I needed to do some heavy work to get those values, so prompting the user was not enough

**Reply by rfon6ngy** [reply]
Here is the extension I was referring to : https://marketplace.visualstudio.com/items?itemName=rioj7.command-variable

**Reply by jfuentesa** [reply]
Ok! Thank you!
